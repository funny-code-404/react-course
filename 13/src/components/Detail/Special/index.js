import PropTypes from "prop-types";

const Special = ({ data }) => (
	data && 
	<div>
		<p>Special:</p>
		<ul>
			{data.map((x) => <li key={x}>{x}</li>)}
		</ul>
	</div>
)

Special.propTypes = {
    data: PropTypes.arrayOf(PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.arrayOf(PropTypes.string)
	]))
}

export default Special;