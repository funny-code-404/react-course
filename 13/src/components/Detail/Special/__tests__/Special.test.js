import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

import Special from "..";

configure({ adapter: new Adapter() });

describe('DETAIL SPECIAL',() => {
    const props = {
        data: [
            'text...'
        ]
    }

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Special {...props} />)
    })
    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('Component is return 1 content', () => {
        expect(wrapper.find('li')).toHaveLength(1);
    });
    it('Component should render correct data content', () => {
        expect(wrapper.find('li').props().children).toEqual(props.data[0]);
    });
    it('Component should render correct after props changing', () => {
        const newdata = 'NEW';
        wrapper.setProps({ data: [ newdata, ]})
        expect(wrapper.find('li').props().children).toEqual(newdata);
    });
})

// npm run test src/components/Detail/Special