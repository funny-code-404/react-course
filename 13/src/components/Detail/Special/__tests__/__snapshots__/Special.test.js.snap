// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DETAIL SPECIAL to Match Snapshot 1`] = `
<div>
  <p>
    Special:
  </p>
  <ul>
    <li
      key="text..."
    >
      text...
    </li>
  </ul>
</div>
`;

exports[`Special to Match Snapshot 1`] = `
<div>
  <p>
    Special:
  </p>
  <ul>
    <li
      key="text..."
    >
      text...
    </li>
  </ul>
</div>
`;
